src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
+// 2. cmd.exe gets a localized name.
+void SettingsLoader::ApplyRuntimeInitialSettings()
 {
-    try
+    // 1.
     {
-        auto settings = LoadDefaults();
-        auto resultPtr = winrt::get_self<CascadiaSettings>(settings);
-        resultPtr->ClearWarnings();
-
-        // GH 3588, we need this below to know if the user chose something that wasn't our default.
-        // Collect it up here in case it gets modified by any of the other layers between now and when
-        // the user's preferences are loaded and layered.
-        const auto hardcodedDefaultGuid = resultPtr->GlobalSettings().DefaultProfile();
-
-        std::optional<std::string> fileData = _ReadUserSettings();
-
-        // Make sure the file isn't totally empty. If it is, we'll treat the file
-        // like it doesn't exist at all.
-        const bool fileHasData = fileData && !fileData->empty();
-        bool needToWriteFile = false;
-        if (fileHasData)
-        {
-            resultPtr->_ParseJsonString(*fileData, false);
-        }
+        const auto preferredPowershellProfile = PowershellCoreProfileGenerator::GetPreferredPowershellProfileName();
+        auto guid = DEFAULT_WINDOWS_POWERSHELL_GUID;
 
-        // Load profiles from dynamic profile generators. _userSettings should be
-        // created by now, because we're going to check in there for any generators
-        // that should be disabled (if the user had any settings.)
-        resultPtr->_LoadDynamicProfiles();
-        try
+        for (const auto& profile : inboxSettings.profiles)
         {
-            resultPtr->_LoadFragmentExtensions();
+            if (profile->Name() == preferredPowershellProfile)
+            {
+                guid = profile->Guid();
+                break;
+            }
         }
-        CATCH_LOG();
 
-        if (!fileHasData)
+        userSettings.globals->DefaultProfile(guid);
+    }
+
+    // 2.
+    {
+        for (const auto& profile : userSettings.profiles)
         {
-            // We didn't find the user settings. We'll need to create a file
-            // to use as the user defaults.
-            // For now, just parse our user settings template as their user settings.
-            auto userSettings{ resultPtr->_ApplyFirstRunChangesToSettingsTemplate(UserSettingsJson) };
-            resultPtr->_ParseJsonString(userSettings, false);
-            needToWriteFile = true;
+            if (profile->Guid() == DEFAULT_COMMAND_PROMPT_GUID)
+            {
+                profile->Name(RS_(L"CommandPromptDisplayName"));
+                break;
+            }
         }
+    }
+}
 
-        try
+// Adds profiles from .inboxSettings as parents of matching profiles in .userSettings.
+// That way the user profiles will get appropriate defaults from the generators (like icons and such).
+// If a matching profile doesn't exist yet in .userSettings, one will be created.
+void SettingsLoader::LayerInboxOntoUser()

```

Comment:
NIT: Name is technically backwards; inbox is "behind" user (if you imagine user is a swiss cheese through which you access settings "below" it)

Reply:
I called it `MergeInboxIntoUser` which I found a lot easier to grasp.